README.md

```diff
@@ -303,12 +305,27 @@ trigram-index = true
 # in memory. This determines how often it will check FS
 # for new or deleted metrics.
 scan-frequency = "5m0s"
+# Control trie index
+#  This index is built as an alternative to trigram index, with shorter indexing
+#  time and less memory usage (around 2 - 5 times). For most of the queries, trie
+#  is faster than trigram. For queries with keyword wrap around by widcards (like
+#  ns1.ns2.*keywork*.metric), trigram indexi performs better. Trie index could
+#  be speeded up by enabling adding trigrams to trie, at the cost of memory usage (
+#  depends on the facotr, the memory usage of trie would be at least doubled).
+trie-index = false
+trie-index-with-trigrams = false
+
 # Maximum amount of globs in a single metric in index
 # This value is used to speed-up /find requests with
 # a lot of globs, but will lead to increased memory consumption
 max-globs = 100
```

Comment:
would this and corresponding code become redundant with introduction of max-metric-globbed and max-metrics-rendered?

Reply:
max-globs is used to control expanding {}, take the example bellow:

```
abc.{1,2,3,4}.{x,y,z}
->
abc.1.x
abc.1.y
...
abc.4.z
```

So this flag max-globs controls how many metrics is expanded to individual metrics, before the actual index query works. Its a flag only used in trigram-index and some special queries of trie-index.

But max-metric-globbed and max-metrics-rendered are used in truncating matched results, it works after the index query. I introduce them to prevent reading too many files and returning too many results that might lead to killing go-carbon. 

They are not necessarily part of trie-index works. But when I was testing trie-index on production queries, I see that trie-index is able to return thousands of metrics on some queries while trigram-index just failed, so I think it might be necessary to  have these control in place for the rendering production test of trie-index.

I admit it's a bit confusing, but they serve a different purpose.